src/content/en/2022/javascript.md

```diff
@@ -16,7 +16,864 @@ featured_stat_2: TODO
+
+{{ figure_markup(
+    caption="The percentage of mobile pages that register at least one paint worklet.",
+    content="0.0013%",
+    classes="big-number",
+    sheets_gid="263104357",
+    sql_file="worklets.sql"
+  )
+}}
+
+With worklets being such niche technologies, it's not surprising that they're not widely used. Paint worklets are an excellent way of offloading expensive processing for generative artwork onto another thread—not to mention a great technique for adding a bit of flair to the user experience. For every 1 million websites, only 13 of them use a paint worklet.
+
+{{ figure_markup(
+    caption="The percentage of mobile pages that register at least one audio worklet.",
+    content="0.0004%",
+    classes="big-number",
+    sheets_gid="263104357",
+    sql_file="worklets.sql"
+  )
+}}
+
+Adoption of audio worklets is even lower: only four in a million websites use it. It will be interesting to see how adoption of these technologies trends over time.
+
+## How is JavaScript delivered?
+
+An equally important aspect of JavaScript performance is how we _deliver_ scripts to the browser, which includes a few common—yet sometimes missed—opportunities for optimization, starting with how we _compress_ JavaScript.
+
+### Compression
+
+[Compression](./compression) is an often-used technique that applies largely to text-based assets, such as HTML, CSS, SVG images, and yes, JavaScript. There are a variety of compression techniques that are widely used on the web that can speed up the delivery of scripts to the browser, effectively shortening the resource load phase.
+
+{{ figure_markup(
+    image="compression-methods.png",
+    caption="Compression of JavaScript by method.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=1805733290&format=interactive",
+    sheets_gid="1841484709",
+    sql_file="compression_method.sql"
+  )
+}}
+
+There are a few compression techniques that can be used to reduce the transfer size of a script, with the [Brotli](https://github.com/google/brotli) (br) method being the [most effective](https://www.smashingmagazine.com/2016/10/next-generation-server-compression-with-brotli/). Despite Brotli's [excellent support in modern browsers](https://caniuse.com/brotli), it's still clear that [gzip](https://www.gzip.org/) is the most preferred method of compression. This is likely due to the fact that many web servers use it as the default.
+
+When something is the default, that default sometimes remains in place rather than being tuned for better performance. Given that only 34% of pages observed are compressing scripts with Brotli, it's clear that there's an opportunity on the table to improve the loading performance of script resources, but it's also worth noting that it _is_ an improvement over last year's adoption at 30%.
```

Comment:
```suggestion
When something is the default, that default sometimes remains in place rather than being tuned for better performance. Given that only 34% of pages observed are compressing scripts with Brotli, it's clear that there's an opportunity on the table to improve the loading performance of script resources, but it's also worth noting that it is an improvement over last year's adoption at 30%.
```